internal/controllers/servicebinding_controller.go

```diff
@@ -39,6 +46,9 @@ type ServiceBindingReconciler struct {
 //+kubebuilder:rbac:groups=servicebinding.io,resources=servicebindings/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=servicebinding.io,resources=servicebindings/finalizers,verbs=update
 //+kubebuilder:rbac:groups=servicebinding.io,resources=clusterworkloadresourcemappings,verbs=get;list;watch
+//+kubebuilder:rbac:groups=apps,resources=deployments;daemonsets;replicasets;statefulsets,verbs=get;list;update;patch
```

Comment:
since the reconciler doesn't need these permissions to operate, we should externalize them as a separate ClusterRole, and aggregate them into the existing [ClusterRole](https://github.com/servicebinding/service-binding-controller/blob/main/config/rbac/role_with_aggregation_rule.yaml) via the `servicebinding.io/controller: "true"` label.